### PR TITLE
chore: revive e2e

### PR DESCRIPTION
**Pull request by myandrienko**
### 🎯 Goal

Bring our e2e tests to at least minimally useful state.

1. Added Playwright browser binaries to cache so that they are not downloaded each time.
2. Removed flaky tests.
3. Removed all browsers except for Chromium - hopefully temporary, but some tests were flaky particularly on WebKit.
4. Playwright will now stop after the first failed tests, so that tests don't hang for 30 mins.
5. Added the whole job to a single concurrency group, because two tests running at once on the same app will conflict.

### CODE REVIEW

**Review by MartinCupela — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -6,27 +6,33 @@ jobs:
...
       - uses: actions/checkout@v3
 
       - name: 💾 Cache Dependencies
         uses: actions/cache@v3
         with:
-          path: ./node_modules
+          path: |
+            ./node_modules
+            ./pw-browsers
           key: ${{ runner.os }}-${{ matrix.node }}-modules-${{ hashFiles('**/yarn.lock') }}
```
Comment (arnautov-anton) [inline .github/workflows/e2e.yml:23]:
```suggestion
          key: ${{ runner.os }}-${{ matrix.node }}-modules-${{ hashFiles('./yarn.lock') }}
```
Let's also get rid of this wildcard, I don't think we care about other lock files at this point in the pipeline.

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -6,27 +6,33 @@ jobs:
...
 
       - name: 🔨 Install Dependencies
-        run: yarn install --frozen-lockfile --ignore-engines --ignore-scripts
-
-      - name: ⚗️ End-to-end tests
         run: |
+          yarn install --frozen-lockfile --ignore-engines --ignore-scripts
           npx playwright install
```
Comment (arnautov-anton) [inline .github/workflows/e2e.yml:28]:
```suggestion
          npx playwright install chromium
```
Let's skip the installation of the other two as those are not needed (for the time being).

Comment (arnautov-anton) [reply .github/workflows/e2e.yml:28]: Other thing (consider it a nit) - would it be faster to move this command to its own step and running it only if the cache-hit is `false`?

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (myandrienko) [reply .github/workflows/e2e.yml:23]: Good point!

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (myandrienko) [reply .github/workflows/e2e.yml:28]: Playwright actually checks if binaries are already up-to-date and only downloads them as necessary. So adding the path with browser binaries to cache is enough.

**Pull request merged**